File: Modules/HumanResource/TrainingManagement/ProgramParticipant/ProgramParticipantForm.cs
using Serenity.ComponentModel;
using System;
using System.Collections.Generic;
using System.ComponentModel;

namespace HRMSoftware.TrainingManagement.Forms;

[FormScript("TrainingManagement.ProgramParticipant")]
[BasedOnRow(typeof(ProgramParticipantRow), CheckNames = true)]
public class ProgramParticipantForm
{
    [DisplayName("Employee Id")]
    public int EmployeeRowId { get; set; }
    [Visible(false)]
    public string EmployeeId { get; set; }
    public string EmployeeName { get; set; }
}

File: Modules/ServerTypes/TrainingManagement/ProgramParticipantForm.ts
import { LookupEditor, StringEditor, PrefixedContext } from "@serenity-is/corelib";
import { initFormType } from "@serenity-is/corelib/q";

export interface ProgramParticipantForm {
    EmployeeRowId: LookupEditor;
    EmployeeId: StringEditor;
    EmployeeName: StringEditor;
}

export class ProgramParticipantForm extends PrefixedContext {
    static formKey = 'TrainingManagement.ProgramParticipant';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!ProgramParticipantForm.init)  {
            ProgramParticipantForm.init = true;

            var w0 = LookupEditor;
            var w1 = StringEditor;

            initFormType(ProgramParticipantForm, [
                'EmployeeRowId', w0,
                'EmployeeId', w1,
                'EmployeeName', w1
            ]);
        }
    }
}